src/components/reusable/table/table-head.ts

```diff
@@ -15,15 +15,33 @@ import styles from './table-head.scss';
 export class TableHead extends LitElement {
   static override styles = [styles];
 
-  override firstUpdated() {
-    this.addEventListener('on-sort-changed', this.handleChildSort as EventListener);
+  @queryAssignedElements()
+  unnamedSlotEls!: Array<HTMLElement>;
+
+  override connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener(
+      'on-sort-changed',
+      this.handleChildSort as EventListener
+    );
+  }
+
+  override disconnectedCallback() {
+    super.disconnectedCallback();
+    this.removeEventListener(
+      'on-sort-changed',
+      this.handleChildSort as EventListener
+    );
   }
 
   private handleChildSort(e: CustomEvent) {
     const sortedColumnKey = e.detail.sortKey;
 
-    // Get all kyn-th elements
-    const allHeaders = Array.from(this.querySelectorAll('kyn-th'));
+    // unnamedSlotEls[0] is the kyn-tr element
+    const parentRow = this.unnamedSlotEls[0];
+
+    // Get all kyn-th children of that kyn-tr element
+    const allHeaders = Array.from(parentRow.querySelectorAll('kyn-th'));
```

Comment:
See comment about using selector in the decorator above. You should be able to loop through `unnamedSlotEls` directly if done correctly.